diff --git a/dist/index.cjs b/dist/index.cjs
index 0af3ffb082855e36ca83eb4caa69ab10b759dc2e..ae27e0833be7f31279a1c89aab4397a6887c2f67 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -931,7 +931,6 @@ function createInitMultipart(boundary) {
 
 /* eslint-disable no-underscore-dangle */
 
-
 class JSONParser extends node_stream.Transform {
   constructor(options = {}) {
     super({ readableObjectMode: true });
@@ -940,13 +939,14 @@ class JSONParser extends node_stream.Transform {
   }
 
   _transform(chunk, encoding, callback) {
-    this.chunks.push(String(chunk)); // todo consider using a string decoder
+    this.chunks.push(chunk); // type: Uint8Array
     callback();
   }
 
   _flush(callback) {
     try {
-      const fields = JSON.parse(this.chunks.join(''));
+      const buffer = Buffer.concat(this.chunks);
+      const fields = JSON.parse(buffer.toString('utf-8'));
       this.push(fields);
     } catch (e) {
       callback(e);
diff --git a/src/parsers/JSON.js b/src/parsers/JSON.js
index 28af758db07390c96671d325bd1f65050666a597..2f7a1fc2fc0affd12c59aadb67c1b6040b8fc5e7 100644
--- a/src/parsers/JSON.js
+++ b/src/parsers/JSON.js
@@ -10,13 +10,14 @@ class JSONParser extends Transform {
   }
 
   _transform(chunk, encoding, callback) {
-    this.chunks.push(String(chunk)); // todo consider using a string decoder
+    this.chunks.push(chunk); // type: Uint8Array
     callback();
   }
 
   _flush(callback) {
     try {
-      const fields = JSON.parse(this.chunks.join(''));
+      const buffer = Buffer.concat(this.chunks);
+      const fields = JSON.parse(buffer.toString('utf-8'));
       this.push(fields);
     } catch (e) {
       callback(e);
