diff --git a/lib/runner/index.js b/lib/runner/index.js
index 87e3b3957619728e3ed1ca61e2d83df1c49f928f..6d5ab905415da0577341c8f5b67d4806adcf7549 100644
--- a/lib/runner/index.js
+++ b/lib/runner/index.js
@@ -68,15 +68,19 @@ function run([, scriptPath, hookName = '', HUSKY_GIT_PARAMS], getStdinFn = get_s
             return 0;
         }
         catch (err) {
-            const noVerifyMessage = [
-                'commit-msg',
-                'pre-commit',
-                'pre-rebase',
-                'pre-push'
-            ].includes(hookName)
-                ? '(add --no-verify to bypass)'
-                : '(cannot be bypassed with --no-verify due to Git specs)';
-            console.log(`husky > ${hookName} hook failed ${noVerifyMessage}`);
+            // We do not want to print this "add --no-verify to bypass" message because that's
+            // literally what some developers do instead of trying to fix the errors.
+
+            // const noVerifyMessage = [
+            //     'commit-msg',
+            //     'pre-commit',
+            //     'pre-rebase',
+            //     'pre-push'
+            // ].includes(hookName)
+            //     ? '(add --no-verify to bypass)'
+            //     : '(cannot be bypassed with --no-verify due to Git specs)';
+            
+            console.log(`husky > ${hookName} hook failed (Please fix the errors listed above and try again)`);
             return err.code;
         }
     });
