diff --git a/android/src/main/java/com/reactnativevosk/VoskModule.kt b/android/src/main/java/com/reactnativevosk/VoskModule.kt
index 0e2b6595b1b2cf1ee01c6c64239c4b0ea37fce19..e0daf712a7f2d2c2dd22e623bac6daba7f6a4ac1 100644
--- a/android/src/main/java/com/reactnativevosk/VoskModule.kt
+++ b/android/src/main/java/com/reactnativevosk/VoskModule.kt
@@ -25,7 +25,9 @@ class VoskModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaMo
 
     // Stop recording if data found
     if (text != null && text.isNotEmpty()) {
-      cleanRecognizer();
+      // Don't auto-stop the recogniser - we want to do that when the user
+      // presses on "stop" only.
+      // cleanRecognizer();
       sendEvent("onResult", text)
     }
   }
