diff --git a/build/rnpm.js b/build/rnpm.js
index d609e53d6146093ef0fabecfce33b21d7b0ea28f..acf932f1b819b04e549c8036bde28400301c73ef 100644
--- a/build/rnpm.js
+++ b/build/rnpm.js
@@ -1265,7 +1265,10 @@
             onPress = _this$props.onPress,
             style = _this$props.style;
         return React__default.createElement(reactNative.TouchableWithoutFeedback, {
-          onPress: onPress
+          onPress: onPress,
+          accessibilityLabel: _this$props.accessibilityLabel,
+          accessibilityRole: 'button',
+
         }, React__default.createElement(reactNative.Animated.View, {
           style: [styles.fullscreen, {
             opacity: this.fadeAnim
@@ -1280,7 +1283,8 @@
   }(React.Component);
 
   Backdrop.propTypes = {
-    onPress: propTypes.func.isRequired
+    onPress: propTypes.func.isRequired,
+    accessibilityLabel: propTypes.string,
   };
   var styles = reactNative.StyleSheet.create({
     fullscreen: {
@@ -1350,6 +1354,7 @@
           style: styles$1.placeholder
         }, React__default.createElement(Backdrop, {
           onPress: ctx._onBackdropPress,
+          accessibilityLabel: this.props.closeButtonLabel,
           style: backdropStyles,
           ref: ctx.onBackdropRef
         }), ctx._makeOptions());
@@ -1981,6 +1986,7 @@
         }), React__default.createElement(MenuPlaceholder, {
           ctx: this,
           backdropStyles: customStyles.backdrop,
+          closeButtonLabel: this.props.closeButtonLabel,
           ref: this._onPlaceholderRef
         }))));
       }
diff --git a/src/index.d.ts b/src/index.d.ts
index 6a9917ffea268379985f70e10221bf52efe865be..81a647ca7570727c99e969e6210479ae05f1af0a 100644
--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -19,6 +19,7 @@ declare module "react-native-popup-menu" {
       backdrop?: StyleProp<ViewStyle>;
       safeArea?: StyleProp<ViewStyle>;
     };
+    closeButtonLabel: string;
     backHandler?: boolean | Function;
     skipInstanceCheck?: boolean;
     SafeAreaComponent?: React.ComponentType<any>;
